source/reading.md

```diff
@@ -607,195 +592,186 @@ using Python with SQLite and PostgreSQL databases.
-by the `select` function allows us to work with data
-stored in databases as if they were just regular data frames; but secretly, behind
-the scenes, `sqlalchemy` is turning your function calls (e.g., `select`)
-into SQL queries! To access the table in the database, we first declare the `metadata` of the table using
-`sqlalchemy` package and then access the table using `select` function from `sqlalchemy` package.
+use the `table` function from the `conn` object. The object returned
+by the `table` function allows us to work with data
+stored in databases as if they were just regular `pandas` data frames; but secretly, behind
+the scenes, `ibis` will turn your commands into SQL queries! 
 
 ```{code-cell} ipython3
-metadata = MetaData(bind=None)
-table = Table(
-    'can_lang', 
-    metadata, 
-    autoload=True, 
-    autoload_with=db
-)
+canlang_table = conn.table("can_lang")
+canlang_table 
+```
+
+```{index} database; count, ibis; count
 ```
 
+Although it looks like we might have obtained the whole data frame from the database, we didn't!
+It's a *reference*; the data is still stored only in the SQLite database. The `canlang_table` object  
+is an `AlchemyTable` (`ibis` is using `sqlalchemy` under the hood!), which, when printed, tells
+you which columns are available in the table. But unlike a usual `pandas` data frame,
+we do not immediately know how many rows are in the table. In order to find out how many
+rows there are, we have to send an SQL *query* (i.e., command) to the data base.
+In `ibis`, we can do that using the `count` function from the table object.
+
 ```{code-cell} ipython3
-query = select([table])
-canlang_data_db = conn.execute(query)
-canlang_data_db
+canlang_table.count()
 ```
 
-```{index} database; fetchall, SQLAlchemy; fetchall
+```{index} execute, ibis; execute
 ```
 
-Although it looks like we just got a data frame from the database, we didn't!
-It's a *reference*; the data is still stored only in the SQLite database. The output 
-is a `CursorResult`(indicating that Python does not know how many rows 
-there are in total!) object.
-In order to actually retrieve this data in Python,
-we use the `fetchall()` function. The
-`sqlalchemy` package works this way because databases are often more efficient at selecting, filtering
-and joining large data sets than Python. And typically the database will not even
+Wait a second...this isn't the number of rows in the database. In fact, we haven't actually sent our 
+SQL query to the database yet! We need to explicitly tell `ibis` when we want to send the query.
+The reason for this is that databases are often more efficient at working with (i.e., selecting, filtering,
+joining, etc.) large data sets than Python. And typically, the database will not even
 be stored on your computer, but rather a more powerful machine somewhere on the
 web. So Python is lazy and waits to bring this data into memory until you explicitly
```

Comment:
```suggestion
web. So Ibis is lazy and waits to bring this data into memory until you explicitly
```
Maybe add a sentence of the benefits of this behavior? "Lazy" might otherwise sound like a bad thing

Reply:
resolved by `ibis`